administrator/components/com_jedchecker/libraries/rules/gpl.php

```diff
@@ -186,14 +186,15 @@ protected function generateRegexp($lines)
 	 */
 	protected function find($file)
 	{
-		// Check the file is empty (i.e. comments-only)
 		$content = php_strip_whitespace($file);
 
-		if (preg_match('#^<\?php\s+$#', $content))
+		// Check the file is empty, comments-only, or nonexecutable
+		if (empty($content) || preg_match('#^<\?php\s+(?:$|(?:die|exit)(?:\(\))?;)#', $content))
```

Comment:
This seems weird to me... why skip files with `die` or `exit` this is not reasonable, but seems like a loop hole, as most of the time obfuscated code will have special functions that can load files and cut away parts of the file and execute the obfuscated code. So skipping these files does not make sense to me. What am I missing here...?

Reply:
It's to avoid false-positives. I've found many extensions have a CHANGELOG.php (or similar) files that starts with `<?php die(); ?>` followed by a text content. Of course, we shouldn't consider these files as a code ones and require full phpDoc comment with license tag.

Reply:
This is too gray for me, as I see a massive loop hole here.

Reply:
Above code just checks that php file is either:
1. Not existing one or not valid php file (`php_strip_whitespace` returns empty string in this case).
2. Just `<?php` w/o other content (with `php_strip_whitespace` it covers cases like `<?php /* Some text content here */`).
3. One of `die` statements (`<?php die;`/`<?php die();`/`<?php exit;`/`<?php exit();`) followed by some content that is not executed because of the previous `die` statement (e.g. see `CHANGELOG.php` in the Akeeba component).

Such excluded files are not *software*, they are just *texts*, and so we shouldn't require them to be distributed under GPL (IMHO).